### PR TITLE
Add password signup and authentication demo

### PR DESCRIPTION
**Pull request by erwan-transmit**
Contains 4 pages:
- signup page to create a new user with a username and password
- login page to login with a username and password
- verify page in charge of doing the code exchange
- home page displayed when the user is logged in

### CODE REVIEW

**Comment by ts-ron** [comment]
Hey @erwan-transmit looks good!
Some comments:

1. I see you are almost aligned to our code structure. One point is we decide to concentrate all backend code under `backend` including the `app.js` file. So if you can rename and consolidate. I think you can keep the separation to `lib` it looks nice and clean, but since its all backend - place it under backend as well?
2. sample.env was moved to the root directory so can remove this from this branch

Some additional comments on the files

**Comment by ts-ron** [comment]
One more comment is the use of tokens on the client side. This would not be our recommendation, and the only case we pulled it in other samples was mostly to show completion of sample flow. I see your setting them as cookies is this something that the flow requires?

**Comment by erwan-transmit** [comment]
Thank you for the review @ts-ron. I will update the PR to match your recommendations.

Regarding the tokens stored in cookies, I am storing them to check if a user is logged in and display the home page when that’s the case. This was also in anticipation of an authentication hub / sso demo where this possibly could have been useful.

Another reason why I stored the tokens provided by Transmit directly is to keep things as simple as possible, for me as a developer, and in the code.

Is there a reason why we don’t recommend storing the tokens client side? My initial guess would be that someone having access to the token would have a larger attack surface since the app is likely to only use a subset of Transmit APIs. But I would be interested in getting more info.

What would you recommend in this case? Creating a session mechanism in the backend capable of delivering and signing tokens and handling refresh tokens?

**Comment by ts-ron** [comment]
@erwan-transmit looks great :) I added some comments regarding remaining reliance on client side access token in `home.html` and the `logout` endpoint - just small touch ups and this is ready for merge 🦾

File: password-authentication/pages/complete.html [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          window.location.replace('/login');
+        }
+
+        // If the code is there, exchange it
+        exchangeAndDisplayIDToken(code);
+      }
+
+      // This functions is triggered the flow is complete, and we can exchange the code for an ID token
+      // For more information see https://developer.transmitsecurity.com/guides/webauthn/cross_device_flows/#step-3-register-credentials
+      async function exchangeAndDisplayIDToken(authCode) {
```
Comment (erwan-transmit) [inline password-authentication/pages/complete.html:41]: This should probably be renamed exchangeIDToken as we don't display it anymore

**Pull request merged**